pkg/controllers/clusterresourceplacement/controller.go

```diff
@@ -722,8 +783,12 @@ func (r *Reconciler) buildPlacementStatus(crp *fleetv1beta1.ClusterResourcePlace
 		return status, nil
 	}
 
-	// TODO collect works and build appliedCondition & resourcePlacementStatus
-	status.PlacementStatuses = buildResourcePlacementStatus(crp, latestSchedulingPolicySnapshot)
+	placementStatuses, appliedCondition, err := r.buildResourcePlacementStatusAndAppliedCondition(ctx, crp, latestSchedulingPolicySnapshot, latestResourceSnapshot)
```

Comment:
A nit: add a comment here to explain that after prev. check the status must be up-to-date.